drivers/serial/Kconfig.stm32

```diff
@@ -7,10 +7,11 @@
 
 menuconfig UART_STM32
 	bool "STM32 MCU serial driver"
-	default n
+	depends on SOC_FAMILY_STM32
 	select SERIAL_HAS_DRIVER
 	select SERIAL_SUPPORT_INTERRUPT
-	depends on SOC_FAMILY_STM32
+	select USE_STM32_LL_USART
```

Comment:
I don't think this was actually required for U(S)ART driver which actually compiled w/o this flag activated.
STM32 serial driver only used LL headers until now, which does not require this flag.
Please re-check and remove

Reply:
I see. Headers are sufficient. Will do.

Reply:
Done.